service/controller/resource/remotewritesecret/create.go

```diff
@@ -0,0 +1,59 @@
+package remotewritesecret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/resourcecanceledcontext"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "ensuring prometheus remotewrite secret")
+	{
+		remoteWrite, err := ToRemoteWrite(obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// fetch current prometheus using the selector provided in remoteWrite resource.
+		prometheusList, err := fetchPrometheusList(ctx, r, remoteWrite)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if prometheusList == nil || len(prometheusList.Items) == 0 {
+			r.logger.Debugf(ctx, "no prometheus found, cancel reconciliation")
+			resourcecanceledcontext.SetCanceled(ctx)
+			return nil
+		}
+
+		for _, current := range prometheusList.Items {
+
+			// Loop over remote write secrets
+			for _, item := range remoteWrite.Spec.Secrets {
+				desired := r.ensureRemoteWriteSecret(item, remoteWrite.ObjectMeta, current.GetNamespace())
+				secret, err := r.k8sClient.K8sClient().CoreV1().Secrets(current.GetNamespace()).Get(ctx, item.Name, metav1.GetOptions{})
+				if apierrors.IsNotFound(err) {
+					r.logger.Debugf(ctx, fmt.Sprintf("creating Secret %#q in namespace %#q", desired.Name, desired.Namespace))
+
+					//if err := r.k8sClient.K8sClient().SetControllerReference(p, &desiredSa, r.Scheme); err != nil {
+					//	return err
+					//}
+					secret, err = r.k8sClient.K8sClient().CoreV1().Secrets(current.GetNamespace()).Create(ctx, &desired, metav1.CreateOptions{})
```

Comment:
I think we need a cleanup mechanism to ensure we do not leave orphaned secrets behind.

When an entry is removed from `remoteWrite.Spec.Secrets` how do we make sure the associated secret is also deleted ?

Reply:
Has been added here:
https://github.com/giantswarm/prometheus-meta-operator/pull/930/files#diff-e16e10a7a44f3be7e69b6787985c26f000958e235184a68d897c45ee6a684c5cR56